server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -204,8 +205,11 @@ public String toString() {
     }
 
     private void onNoLongerPrimary(Exception failure) {
-        final boolean nodeIsClosing = failure instanceof NodeClosedException ||
-            (failure instanceof TransportException && "TransportService is closed stopped can't send request".equals(failure.getMessage()));
+        final boolean nodeIsClosing = failure instanceof NodeClosedException
```

Comment:
Perhaps use `ExceptionsHelper.unwrapCause`? `SendRequestTransportException` is implementing `ElasticsearchWrapperException`.